docker/build-tools/Dockerfile

```diff
@@ -44,31 +44,30 @@ ENV GOIMPORTS_VERSION=v0.1.0
 ENV BENCHSTAT_VERSION=9c9101da8316
 ENV GH_VERSION=2.27.0
 ENV GOLANG_PROTOBUF_VERSION=v1.30.0
-ENV GOLANG_GRPC_PROTOBUF_VERSION=v1.2.0
+ENV GOLANG_GRPC_PROTOBUF_VERSION=v1.3.0
 # When updating the golangci version, you may want to update the common-files config/.golangci* files as well.
 ENV GOLANGCI_LINT_VERSION=v1.51.2
 ENV HADOLINT_VERSION=v2.12.0
-ENV HELM3_VERSION=v3.11.2
+ENV HELM3_VERSION=v3.11.3
 ENV HUGO_VERSION=0.111.3
 ENV JB_VERSION=v0.3.1
 ENV JSONNET_VERSION=v0.15.0
 ENV JUNIT_MERGER_VERSION=adf1545b49509db1f83c49d1de90bbcb235642a8
-ENV K8S_CODE_GENERATOR_VERSION=1.27.0
+ENV K8S_CODE_GENERATOR_VERSION=1.27.1
 ENV K8S_TEST_INFRA_VERSION=2acdc6800510dd422bfd2a5d8c02aedc19d15f8d
 ENV KIND_VERSION=v0.18.0
-ENV KUBECTL_VERSION=1.26.3
+ENV KUBECTL_VERSION=1.27.1
 ENV PROTOC_GEN_GRPC_GATEWAY_VERSION=v1.8.6
 ENV PROTOC_GEN_SWAGGER_VERSION=v1.8.6
 ENV PROTOC_GEN_VALIDATE_VERSION=v0.9.1
 ENV PROTOC_VERSION=22.1
 ENV PROTOLOCK_VERSION=v0.14.0
-ENV PROTOTOOL_VERSION=v1.10.0
 ENV SHELLCHECK_VERSION=v0.9.0
 ENV SU_EXEC_VERSION=0.2
-ENV YQ_VERSION=4.33.2
+ENV YQ_VERSION=4.33.3
 ENV KPT_VERSION=v0.39.3
-ENV BUF_VERSION=v1.13.1
-ENV GCLOUD_VERSION=425.0.0
+ENV BUF_VERSION=v1.17.0
```

Comment:
@ericvn we need to see if this works on mac or if it is still broken

Reply:
I'll build and run `make gen` again to verify Mac behavior. It looks like from my earlier comments the pipeline also broke so shouldn't have been a Mac only issue.

Reply:
@jacob-delgado I was able to do a `make gen` with the new image I built (failed on the first `buf generate` before):
```
HEAD is now at 51f7666f updating header presence explanation with example (#2762)
buf generate --config tools/proto/buf.yaml --path operator/pkg/ --output operator  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/dns/ --output pkg  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/test/echo --output pkg  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/workloadapi --output pkg  --template tools/proto/buf.golang.yaml
rm manifests/charts/istiod-remote/templates/*
```